### ISSUE TITLE
Background builds with terminal multiplexer

### ISSUE THREAD

**Issue by puneetse** [clearlinux/autospec#391]
It might be a nice feature if autospec could detect a terminal multiplexer, like screen or tmux, if available and leverage it when kicking off builds.

This could allow long running builds to not be interrupted if the autospec process is interrupted due to SSH connection drop and give the option to monitor build progress in realtime by attaching to the session (I tail build.log as a habit anyways).

Autospec could further maintain the state of build, so if it was interrupted it could reattach to the running build instead of restarting.

The obvious alternative would be to start a new multiplexed session before starting autospec, but that takes some remembering :)

**Comment by bryteise** [comment]
So you would want to have autospec be able to exec the multiplexer which then starts the autospec run, potentially with a window tailing the log as the build progresses?

**Comment by puneetse** [comment]
@bryteise precisely. Feel free to call it out of scope since there are alternatives but I think it would be a neat integration that doesn't have to create hard dependency.

If automatically showing a tail window is a problem, even just printing the command to attach to the window in a separate terminal would work.

**Comment by bryteise** [comment]
I'm not really sure I would want to involve autospec in knowing about these kinds of things. I could think of a number of programs that I'd say could be given this kind of feature and I think it just comes down to the person running the tooling using what they want to get this behavior.

**Comment by phmccarty** [comment]
I think this type of feature is more appropriate for an autospec *wrapper*, not for autospec itself.

**Comment by phmccarty** [comment]
But also consider: if any additional functionality from autospec is desired to ease the development of such a wrapper, then I think those changes would be worth considering.
